.github/workflows/lint.yml

```diff
@@ -0,0 +1,28 @@
+name: android lint and ktlintCheck
```

Comment:
let's keep the [naming convention](https://github.com/logto-io/logto/blob/master/.github/workflows/main.yml) of workflows. `Main` will be good enough for lint and build (in the future) also the filename